blobstore-fs/src/main.rs

```diff
@@ -0,0 +1,676 @@
+//! blobstore-fs capability provider
+//!
+//!
+
+use log::{error, info};
+#[allow(unused_imports)]
+use serde::Deserialize;
+use std::time::SystemTime;
+#[allow(unused_imports)]
+use std::{
+    collections::HashMap,
+    fs::OpenOptions,
+    fs::{metadata, read, read_dir, remove_file, File},
+    io::{BufReader, Write},
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use tokio::sync::RwLock;
+use wasmbus_rpc::provider::prelude::*;
+use wasmbus_rpc::Timestamp;
+use wasmcloud_interface_blobstore::*;
+mod fs_utils;
+pub use fs_utils::all_dirs;
+
+#[allow(unused)]
+const CAPABILITY_ID: &str = "wasmcloud:blobstore";
+#[allow(unused)]
+const FIRST_SEQ_NBR: u64 = 0;
+
+// main (via provider_main) initializes the threaded tokio executor,
+// listens to lattice rpcs, handles actor links,
+// and returns only when it receives a shutdown message
+//
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
All of our capability providers now create/initialize a tracing subscriber here. It should be an easy set of copy/paste, as you can see an example here https://github.com/wasmCloud/capability-providers/blob/main/blobstore-s3/src/main.rs#L18

Reply:
Done, but what does this do?